### PR TITLE
Password check

### PR DESCRIPTION
**Pull request by Regggis**
Only bonus job to do!

### CODE REVIEW

**Comment by ziobron** [comment]
✅ CI Passed - Task: password-check
🏅 Granted 8 XP for @Regggis

**Comment by ziobron** [comment]
⏱ 3 XP extra for @Regggis for on-time delivery

**Comment by ziobron** [comment]
✅ CI Passed - Task: password-check-bonus
🏅 Granted 4 XP for @Regggis

File: homework/password-check/validation.cpp [hunk 1]
```diff
@@ -1,2 +1,68 @@
...
+            return "Password needs to have at least one uppercase letter";
+        }
+        case ErrorCode::PasswordsDoNotMatch : {
+            return "Passwords do not match";
+        }
+        default : {
+            return "Out of range";
+        }
+    }
+    return "Out of range";
```
Comment (nauka-programowania-MA) [inline homework/password-check/validation.cpp:27]: Choose one of them

File: homework/password-check/validation.cpp [hunk 2]
```diff
@@ -1,2 +1,68 @@
...
+}
+bool doPasswordsMatch(std::string password1, std::string password2) {
+    if (password1 == password2) {
+        return true;
+    }
+    return false;
+}
+ErrorCode checkPasswordRules(std::string password) {
```
Comment (nauka-programowania-MA) [inline homework/password-check/validation.cpp:35]: Add new line between functions

File: homework/password-check/validation.cpp [hunk 3]
```diff
@@ -1,2 +1,68 @@
...
+        case ErrorCode::PasswordsDoNotMatch : {
+            return "Passwords do not match";
+        }
+        default : {
+            return "Out of range";
+        }
+    }
+    return "Out of range";
+}
+bool doPasswordsMatch(std::string password1, std::string password2) {
```
Comment (nauka-programowania-MA) [inline homework/password-check/validation.cpp:29]: Pass by const refrence. Try always avoid copy.

File: homework/password-check/validation.cpp (hunk 2 above)
Comment (nauka-programowania-MA) [inline homework/password-check/validation.cpp:35]: Pass by const refrence. Try always avoid copy.

**Review by nauka-programowania-MA — APPROVED** [review]

**Pull request closed without merging**